lib/invoca/utils/exceptions.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Invoca
+  module Utils
+    class << self
+      # Yields and rescues any exceptions given in `exception_classes`, retrying the given number of times.
+      # The final retry does not rescue any exceptions.
+      # The try number (0..retries) is yielded as a block param.
+      #
+      # @param [Class or Array(Class)] exception_classes - exception()s) to rescue
+      # @param [Integer] retries: - 1+ count of retries (1 retry = up to 2 tries total)
+      # @param [Proc] before_retry - optional proc which is called before each retry, with the exception passed as a block param
+      # @return the value from yield
+      def retry_on_exception(exception_classes, retries:, before_retry: nil)
```

Comment:
I know this probably is just a direct copy over, but I have a couple questions for the new definition.

1. should we default `retries` to 1 as a sane default?
2. should `exception_classes` default to to a catch all `StandardError`?
3. if the answer to 1 and 2 are no, would it be better to make all the arguments kwargs?

Reply:
> 1. should we default retries to 1 as a sane default?

I often don't go for defaults when there are no callers--on the reasoning that explicit is better--but I suppose it would be pretty clear that `retry_on_exception` retries just once.

> 2. should exception_classes default to to a catch all StandardError?

Nah, in all the case I've used this method or a similar one, I've never used `StandardError`. It's always been more targeted than that.

I did consider making the exception classes argument `*exception_classes` so you could provide a list without `[` `]` around them. But I didn't do that, because it's pretty common for the exception list to be in a big constant, and if I use use `*` to scoop them up, that forces the caller to use `*` to splat them out.